python/lsst/pipe/tasks/functors.py

```diff
@@ -778,7 +785,7 @@ def __init__(self, col, calib=None, **kwargs):
         if calib is not None:
             self.fluxMag0 = calib.getFluxMag0()[0]
```

Comment:
If we want to remove `calib`, we should first document that this is an unused argument in the docstring, remove this line and instead raise a deprecation warning here. I'd follow a pattern similar to these lines: https://github.com/lsst/pex_config/blob/aa9e307f6b0b52071272d5b1404e053a96c8fdb0/python/lsst/pex/config/config.py#L1543-L1546

Reply:
How does this revision (just pushed) look to you? Future deprecation ticket: [DM-39914](https://jira.lsstcorp.org/browse/DM-39914).

Reply:
This looks fine, as long as you're sure `getFluxMag0` exists. My GitHub search returns no implementation of it. If that's the case, I'd remove it right away.

Reply:
And either way, the warning should come first before it executes the code in the `if` block.

Reply:
As this hasn't been deprecated officially yet, people may still want to use this to access older data. I realize that this may not be something we need to support here, but regardless, my feeling was that it costs us nothing to keep it in for now just in case.